src/ApplicationManager.cpp

```diff
@@ -35,32 +35,44 @@ void ApplicationManager::loop() {
     int nTicksTaken = 0;
     double timeTakenOverNTicks = 0.0;
 
+    std::chrono::milliseconds now_time;
+    std::chrono::milliseconds last_call_time = now_time;
+    std::chrono::milliseconds last_fps_count_time = now_time;
 
-    std::chrono::milliseconds ms = std::chrono::duration_cast< std::chrono::milliseconds >(
-        std::chrono::system_clock::now().time_since_epoch()
-    );
+    bool ok = true;
 
-    std::chrono::milliseconds last_call_time = ms;
+    int cyclesInThisSecond = 0;
 
-    bool ok = true;
+    int lastFPS = 0;
     while(ok) {
 
-        std::chrono::milliseconds ms = std::chrono::duration_cast< std::chrono::milliseconds >(
+        now_time = std::chrono::duration_cast< std::chrono::milliseconds >(
             std::chrono::system_clock::now().time_since_epoch()
         );
-        std::chrono::milliseconds now_time = ms;
-
         auto diff = now_time - last_call_time;
-
         auto timeDiff =std::chrono::milliseconds(16); //a little over 60hz
-
         if(diff > timeDiff) {
-
             this->runOneLoopCycle();
+
+            std::chrono::milliseconds afterOneCycleTime = std::chrono::duration_cast< std::chrono::milliseconds >(
+                    std::chrono::system_clock::now().time_since_epoch());
+
+            cyclesInThisSecond++;
+
+            auto fps_diff = now_time - last_fps_count_time;
+            auto fps_timestep =std::chrono::milliseconds(200); // one second
+            if(fps_diff > fps_timestep) {
+
+                std::cout << "FPS: " << cyclesInThisSecond << std::endl;
+                lastFPS = cyclesInThisSecond;
+                cyclesInThisSecond=0;
+                last_fps_count_time = now_time;
+            }
+            ai::interface::Input::setFps(lastFPS*5);
```

Comment:
If would only set it in the interface everytime we actually update the value (so within the if Statement). Now you're sending data that's really generated at 5hz at 60hz to the interface.